packages/astro/src/cli/info/index.ts

```diff
@@ -12,31 +15,82 @@ interface InfoOptions {
 	flags: Flags;
 }
 
+export const VERSION_PREFIX_REGEX = /[\^~<>=]/g;
+
+type MinimalPackageJSON = {
+	version: string;
+	dependencies: Record<string, string>;
+}
+
+function getVersion(packageJsonPath: string, dependency?: string): string | undefined {
+	try {
+		const resolvedPath = createRequire(import.meta.url).resolve(packageJsonPath);
+		
+		const packageJSON = JSON.parse(readFileSync(resolvedPath, "utf-8")) as MinimalPackageJSON;
+		
+		if (dependency) {
+			if (!packageJSON.dependencies) return undefined;
+			
+			const depVersion = packageJSON.dependencies[dependency];
+			
+			if (!depVersion) return undefined;
+			
+			const sanitizedDepVersion = depVersion.replaceAll(VERSION_PREFIX_REGEX, "");
+			
+			return sanitizedDepVersion;
+		} else {
+			return packageJSON.version;
+		}
+	} catch {
+		return undefined;
+	}
+}
+
 export async function getInfoOutput({
 	userConfig,
 	print,
+	root
 }: {
 	userConfig: AstroUserConfig | AstroConfig;
 	print: boolean;
+	root: string;
 }): Promise<string> {
+	const astroViteVersion = getVersion("../../../package.json", "vite");
+	const userViteVersion = getVersion(path.join(root, "package.json"), "vite");
```

Comment:
Not a reliable source. This will show the version range for both. The actually installed version of Vite might be neither of those due to version resolution or even outside the ranges if there is an override. We need to call the package manager to know the actually installed version of packages.

I'd say even more important than the version of Vite is whether there are more than one version installed, since that would break everything that relies on global state.

All the package managers we support have the `why` command, so we could invoke `npm/pnpm/yarn/bun why` and get the result with some small parsing (I think all of them even have json outputs)